python/rmm/rmm/tests/test_rmm.py

```diff
@@ -1002,3 +1002,15 @@ def test_invalid_logging_level(level):
         rmm.set_flush_level(level)
     with pytest.raises(TypeError):
         rmm.should_log(level)
+
+
+def test_available_device_memory():
+    from rmm._lib.memory_resource import available_device_memory
+
+    initial_memory = available_device_memory()
+    device_buffer = rmm.DeviceBuffer.to_device(  # noqa: F841
+        np.zeros(10000, dtype="u1")
+    )
+    final_memory = available_device_memory()
+    assert initial_memory[1] == final_memory[1]
+    assert initial_memory[0] > final_memory[0]
```

Comment:
I wonder if this test will be safe to run in parallel. We may need some kind of marker on this so it runs by itself, or perhaps we just need to be ready to disable the last part of the test (final free < initial free) if it seems to be flaky in CI. I still think the other part (final total == initial total) is useful even if we have to skip the latter part as flaky.

Reply:
This is a good point. I don't think that "available memory is always less immediately after allocating memory" should be a requirement of the `available_device_memory` query that we want to enforce via testing.

I think `available_device_memory` is guaranteed to return a non-negative integer, and it should not throw if the cuda context and device are in a good state. I don't think it can guarantee much else.

Reply:
Done 👍